frost-core/src/frost.rs

```diff
@@ -238,10 +240,19 @@ where
     #[cfg_attr(feature = "internals", visibility::make(pub))]
     pub fn binding_factor_preimages(
         &self,
+        group_public: &VerifyingKey<C>,
         additional_prefix: &[u8],
     ) -> Vec<(Identifier<C>, Vec<u8>)> {
         let mut binding_factor_input_prefix = vec![];
 
+        // The length of a serialized verifying key of the same cipersuite does
+        // not change between runs of the protocol, so we don't need to hash to
+        // get a fixed length.
+        //
+        // TODO: when serde serialization merges, change this to be simpler?
+        binding_factor_input_prefix
+            .extend_from_slice(<C::Group>::serialize(&group_public.element).as_ref());
```

Comment:
I don't think serde helps here, it helps more for serializing structs. But you can use `group_public.to_bytes()` which should simplify it.

Reply:
yeah that's much better